pkg/builder/service_builder.go

```diff
@@ -33,6 +33,7 @@ func (b *Builder) BuildService(mariadb *mariadbv1alpha1.MariaDB, key types.Names
 		metadata.NewMetadataBuilder(key).
 			WithMariaDB(mariadb).
 			WithAnnotations(opts.Annotations).
+			WithLabels(opts.Selectorlabels).
```

Comment:
I think it would be easier to do the following:
```go
	selectorLabels :=
		labels.NewLabelsBuilder().
			WithMariaDBSelectorLabels(mariadb).
			WithLabels(opts.Selectorlabels).
			Build()
	objMeta :=
		metadata.NewMetadataBuilder(key).
			WithMariaDB(mariadb).
			WithAnnotations(opts.Annotations).
			WithLabels(selectorLabels).
			Build()

```
What do you think?

Reply:
Looks good to me. Updated the solution